system/lib/wasmfs/paths.h

```diff
@@ -11,6 +11,11 @@
 
 namespace wasmfs {
 
+struct ParsedPath {
+  std::optional<Directory::Handle> parent;
```

Comment:
Surely all files have a parent.. even those that live in the root ?

Reply:
Not files that have been unlinked and are only kept alive by the open file table!

Reply:
Also the root itself has no parent.

Reply:
Oh yeah, I guess an unlinked file wouldn't be returned by path parsing anyway.

Reply:
Hmm, it could be unlinked by another thread after parsing, though...